src/main/resources/static/javascript/mainApp/general.controller.js

```diff
@@ -658,6 +658,35 @@
          * The file is retrieved from the html input with id 'upload'.
          */
         $scope.readTextFile = function (inputFile) {
+            const type = inputFile.type;
+            if (!$scope.verifyImportFileType(inputFile)) {
+                $scope.launchDynamicModal(
+                    Message.Title.INVALID_FILE,
+                    "File must be a .txt file."
+                );
+                return;
+            }
+            if (!$scope.verifyImportFileSize(inputFile)) {
+                $scope.launchDynamicModal(
+                    Message.Title.INVALID_FILE,
+                    "File is too large. Maximum file size: 5MB"
+                );
+                return;
+            }
+            if (!$scope.verifyImportFileNameSize(inputFile)){
+                $scope.launchDynamicModal(
+                    Message.Title.INVALID_FILE,
+                    "File name has too many characters. Maximum character amount: 50"
+                );
+                return;
+            }
+            if (!$scope.verifyImportFileName(inputFile)){
+                $scope.launchDynamicModal(
+                    Message.Title.INVALID_FILE,
+                    "File name has illegal characters."
+                );
+                return;
+            }
```

Comment:
Can you use change this to a long else-if chain to avoid using a return statement? Could you also move the all invalid file messages into the app.constants.js file too?

Reply:
Let's just keep this style. The const for the type at the top doesn't appear to be used, though.

Reply:
Thanks Frank.  I was going to mention this too.  Jorden, academics used to teach that methods should have one exit, at the bottom.  But the craft has evolved.  Guards were introduced.  Avoiding complexity by exiting early has become an accepted practice.